src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -92,6 +92,23 @@ Copyright (c) .NET Foundation. All rights reserved.
     </_GenerateRestoreGraphProjectEntryInputProperties>
   </PropertyGroup>
 
+  <ItemDefinitionGroup Condition="'$(ManagePackageVersionsCentrally)' == 'true' And '$(RestoreEnableGlobalPackageReference)' != 'false'">
+    <GlobalPackageReference>
+      <!--
+        Default central package references to only consume Analyzers and Build assets of a package.  This helps ensure that the package
+        assets are not passed to the compiler or copied to the output directory.  Having a compile-time reference to a package for all
+        projects in a tree is not recommended.  Instead, central package references should just extend the build.
+      -->
+      <IncludeAssets Condition="'$(RestoreEnableGlobalPackageReferenceIncludeAssetsBuildAnalyzers)' != 'false'">Analyzers;Build;BuildMultitargeting;BuildTransitive</IncludeAssets>
+      <!--
+        Default global package references to have all assets private.  This is because central package references are generally packages
+        that provide versioning, signing, etc and should not flow to downstream dependencies.  Also, central package references are
+        already referenced by every project in the tree so they don't need to be transitive.
+      -->
+      <PrivateAssets Condition="'$(RestoreEnableGlobalPackageReferencePrivateAssetsAll)' != 'false'">All</PrivateAssets>
```

Comment:
Do we even want to make this overridable? 

This is a very particular/specific feature. We're likely to run into fewer problems if we just don't allow this to be a thing.

Reply:
I agree. We can remove them but we should not limit global packages item type to only build dependencies. It's up to those who use it.

Even if it's unsupported, proper documentation should tell users that they can override the item definition (_to what they want from the package reference as they do today_). But warn them that it's an unsupported scenario.

_I don't know why it's an unsupported scenario since all it does is add them to package reference items, which is supported._

See https://github.com/NuGet/NuGet.Client/pull/4680#discussion_r920697844

Even @dotnet org folks can benefit from this.

Reply:
For now I'm going to not allow any customizations to the assets selection.  I agree that if a user customizes those values, it could lead to issues where things aren't working as intended.  If a user wants to customize how global packages are used, they'll need to do it entirely in a custom way so they know that support won't be provided directly from the NuGet team.  If later on we get a lot of requests to make it customizable, we can add that later.